nx/lib/nx/binary_backend.ex

```diff
@@ -630,9 +630,15 @@ defmodule Nx.BinaryBackend do
 
   @impl true
   def select(out, %{shape: {}} = pred, on_true, on_false) do
+    on_false =
+      on_false |> broadcast_data(out.shape) |> binary_to_binary(on_false.type, out.type, & &1)
+
+    on_true =
+      on_true |> broadcast_data(out.shape) |> binary_to_binary(on_true.type, out.type, & &1)
```

Comment:
The goal was to compute the broadcast and the type handling only for the value we are interested. Can we move this back to the conditional? Or maybe do this:

```
result = if scalar(...), do: on_true, else: on_false
result |> broadcast_data() |> binary_to_binary() |> from_binary()
```